dulwich/repo.py

```diff
@@ -1305,6 +1305,58 @@ def stage(self, fs_paths):
                     index[tree_path] = index_entry_from_stat(st, blob.id, 0)
         index.write()
 
+    def unstage(self, relpaths: List[bytes] = []):
+        """unstage specific file in the index
+        Args:
+          repo: dulwich Repo object
+          paths: a list of file to unstage, relative to the repository path
+        """
+
+        from dulwich.index import IndexEntry
+        from dulwich.porcelain import remove
+
+        index = self.open_index()
+        try:
+            tree_id = self[b'HEAD'].tree
+        # no head mean no commit in the repo
+        except KeyError:
+            remove(self, paths=[os.path.join(self.path, path.decode()) for path in relpaths], cached=True)
```

Comment:
porcelain is low level code, please don't add circular imports from it into dulwich.repo. It should be the other way around - dulwich.porcelain.remove() calls Repo.unstage.

Reply:
updated in the new commit